### PR TITLE
Add sdk info

### PR DESCRIPTION
**Pull request by pablisco**
Adds the requirements to provide with sdk values to be added to an instance and groups platform dependencies into a single object.

### CODE REVIEW

**Comment by pablisco** [comment]
[Add Android client SDK and version as request headers so that we know what SDKs our customers are using](https://trello.com/c/qdJAQGH1/677-add-android-client-sdk-and-version-as-request-headers-so-that-we-know-what-sdks-our-customers-are-using)

File: library-core/src/main/kotlin/com/pusher/platform/BaseClient.kt [hunk 1]
```diff
@@ -204,3 +215,7 @@ class BaseClient(
 
     private fun absolutePath(path: String): String = "$baseUrl/$path".replaceMultipleSlashesInUrl()
 }
+
+internal fun Request.Builder.add(headers: Headers) = headers.forEach { (name, values) ->
```
Comment (mdpye) [inline library-core/src/main/kotlin/com/pusher/platform/BaseClient.kt]: Is this used?

Comment (pablisco) [reply library-core/src/main/kotlin/com/pusher/platform/BaseClient.kt]: It's used in `BaseSubscription`:
https://github.com/pusher/pusher-platform-android/pull/26/files#diff-8246ed0ce0e2bf0385b2e37bb3124865R44

Could move it there or have a specific extention file for `okhttp`

Comment (mdpye) [reply library-core/src/main/kotlin/com/pusher/platform/BaseClient.kt]: Oh, I see. I was confused by how the syntax (and self pointer) is moving around there

**Pull request merged**